JSIL.Libraries/Includes/Bootstrap/Text/Classes/System.Char.js

```diff
@@ -79,6 +87,36 @@ JSIL.ImplementExternals("System.Char", function ($) {
     }
   );
 
+  $.Method({ Static: true, Public: true }, "ConvertToUtf32",
+    new JSIL.MethodSignature($.Int32, [$.String, $.Int32], []),
+    function ConvertToUtf32(s, i) {
+      // If codePointAt is available, use it, but it isn't
+      // widespread enough to rely on yet.
+      if (s.codePointAt)
+        return s.codePointAt(i);
+      // Otherwise, do shifting arithmetic for surrogate pairs.
+      var hs = s.charCodeAt(i);
```

Comment:
Sorry to take so long to get back to you on this, but there's an implementation of this surrogate pair workaround in the text encoding code already, named ```$jsilcore.charCodeAt``` and ```$jsilcore.fromCharCode```.